src/views/SignUpView.vue

```diff
@@ -1,36 +1,79 @@
 <template>
   <main>
+    <error-toast v-show="isToastShown" text="Ooops, something went wrong. Try again." />
+
     <div class="py-12">
       <h2 class="mb-8 text-3xl text-center font-bold tracking-wider">Sign up</h2>
       <the-form>
         <template #inputs>
           <div class="flex flex-col px-6">
-            <the-input type="text" placeholder="User name"></the-input>
-            <the-input type="password" placeholder="Password"></the-input>
-            <the-input type="password" placeholder="Repeat password"></the-input>
-            <the-input type="email" placeholder="Email"></the-input>
+            <the-input
+            v-model="userData.name"
+            type="text"
+            placeholder="User name"
+            :conditions="isFormSubmitted && errorMessages.nameError !== ''"
+            :errorText="errorMessages.nameError"
+            >
+            </the-input>
+            <the-input
+            v-model="userData.passOne"
+            type="password"
+            placeholder="Password"
+            :conditions="isFormSubmitted && errorMessages.passOneError !== ''"
+            :errorText="errorMessages.passOneError"
+            >
+            </the-input>
+            <the-input
+            v-model="userData.passTwo"
+            type="password"
+            placeholder="Repeat password"
+            :conditions="isFormSubmitted && errorMessages.passTwoError !== ''"
+            :errorText="errorMessages.passTwoError"
+            >
+            </the-input>
+            <the-input
+            v-model="userData.email"
+            type="email"
+            placeholder="Email"
+            :conditions="isFormSubmitted && errorMessages.emailError !== ''"
+            :errorText="errorMessages.emailError"
+            >
+            </the-input>
```

Comment:
I would map userData you use in `v-model` and create a computed property which would be an array of objects like so:
```js
computed: {
  inputs() {
    return [{
      model: userData.name,
      type: "text",
      placeholder,
      condition: this.FormSubmitted && this.errorMessages.nameError !== ''",
      error: this.errorMessages.nameError
    }]
  }
}
```
Then in template:
```jsx
<the-input 
  v-for="(input, index) in inputs"
  :key="signin-`${index}`"
  v-model="input.model"
  :type="input.type"
  :conditions="input.condition",
  :errorText="input.error"
/>

Reply:
Good idea, I hadn't thought like that, thank you 🙃

**Corrected**